apertium-eng.eng.rlx

```diff
@@ -329,6 +344,33 @@ SELECT V IF (-1 (prn pers)) ;
 SELECT SUB:1 Be IF (1 Ing) (0/* Have) ;
 SELECT Ger IF (-1/* Be) (1 To) ;
 
+SELECT Adv IF (0 Pr) (-1 Vblex) ;
+#Example (on):
+#He talked on.
+
+SELECT Cnjadv IF (0 Pr) (1 Det) (2 N) ;
+#Example (for):
+#Please get fish for the cat.
+
+SELECT Adv IF (0 N) (1 Supadj) ;
+#Example (smallest)
+#Canada is the second largest country in the world.
```

Comment:
What other words are ambiguous between `adv` and `n`?  Will this rule work for those too?  Or will it at least not select the wrong reading?

Reply:
I'm pretty sure that all words that match this pattern adverb, superlative adjective, will work.

Reply:
I'm pretty sure all adverbs that match this pattern (Adv) (Superlative Adjective) should work.

`third largest`

Reply:
I don't think any other adverbs match this pattern besides number-based adverbs, so it shouldn't cause any trouble.

Reply:
Yes, but some nouns (that might otherwise be ambiguous with adverbs) might match this pattern.  Could you answer this question?

> What other words are ambiguous between `adv` and `n`?

Reply:
Here are some words that are both nouns and adverbs:
 
Down: `down (feathers)`, `rode the lift up and skied down`
Fast: `Lent and Ramadan are two fasts`, `The clock is running fast`
Firm: `A banking firm`, `You must stand firm`

Reply:
This rule will select the wrong thing for "They named the banking firm greatest in the city.", right?

Reply:
That's true.  I'll make it so that the adverb/noun must be proceeded by a determiner.

Reply:
Hm, that might not work, though, if in the case 
`They named the firm greatest in the country`.

Reply:
This is probably safe now I guess..